share/ipld/nmt_adder.go

```diff
@@ -151,6 +159,18 @@ func (a *ProofsAdder) VisitFn() nmt.NodeVisitorFn {
 	return a.visitInnerNodes
 }
 
+// Purge removed proofs from ProofsAdder allowing GC to collect the memory
+func (a *ProofsAdder) Purge() {
+	if a == nil {
+		return
+	}
+
+	a.lock.RLock()
+	defer a.lock.RUnlock()
+
+	a.proofs = nil
```

Comment:
nit: might be worth using `map.Clear`(or an old way to clear map) to help GC a bit

Reply:
Not sure what is map.Clear, it is something I'm not aware of. The best way to help GC is to remove links to object, so it does exactly that 🙂

Reply:
By helping GC I mean freeing memory in-place rather than deferring this to GC

Reply:
Is there a way to forcefully free up space for reference type? Asking because I'm not aware of it. Deleting items from map for example, will not remove the map, it will just mark items as soft-deleted

Reply:
Doesnt answer the question, but related https://stackoverflow.com/a/70808604